### ISSUE TITLE
PFFFT compile failure in Windows ARM64

### ISSUE THREAD

**Issue by MikuAuahDark** [kcat/openal-soft#979]
When compiling for Windows ARM64 using MSVC, this compile error occurs:
```
D:\omitted\openal-soft\common\pffft.cpp(308,1): error C1021: invalid preprocessor command 'warning' [D:\omitted\openal-soft\build\alcommon.vcxproj]
```

It looks like PFFFT does not consider `_M_ARM64` as ARM64 nor NEON-capable and MSVC does not define `__ARM_NEON` at all. Adding `_M_ARM64` define check solves the issue.

```diff
diff --git a/common/pffft.cpp b/common/pffft.cpp
index 3b74dcfd..fe09390c 100644
--- a/common/pffft.cpp
+++ b/common/pffft.cpp
@@ -189,7 +189,7 @@ force_inline v4sf vswaphl(v4sf a, v4sf b) noexcept
 /*
  * ARM NEON support macros
  */
-#elif defined(__ARM_NEON) || defined(__aarch64__) || defined(__arm64)
+#elif defined(__ARM_NEON) || defined(__aarch64__) || defined(__arm64) || defined(_M_ARM64)
 
 #include <arm_neon.h>
 using v4sf = float32x4_t;
```

As for the actual error, MSVC does not support `#warning` directive. The closest is `#pragma message` but it's not 100% equivalent.

**Comment by kcat** [comment]
Should be fixed with commit 4a74b22e2c80ceba6ef1383715b457653e463a6c.

**Comment by MikuAuahDark** [comment]
Yep, that does it. Thanks.
